lcls_naming_tool/lcls_naming_tool.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env python3
+
+'''
+    LCLS Naming Tool
+
+    A tool for checking the form and content (but not meaning) of PV names with
+    respect to the LCLS naming convention.
+
+    A valid PV name is in the format FFFFF:GGG:CCC:NN:XXXX, where GGG and NN are
+    optional and can be omitted.
+
+    This script only checks the validity of PV names (not devices).
+
+    Instructions:
+    1. Set $ chmod 775 lcls_naming_tool.py to make the file executable.
+
+    2. To check if a PV name is valid type pipe in the text like so 
+            $ echo "XCS:DG3:GCC:02:PCTRLSPRBCK" | ./lcls_naming_tool.py
+    
+    3. To check if a list of PV names is valid pipe in the file name like so 
+            $ cat pvlist.txt | ./lcls_naming_tool.py
+'''
+
+
+import sys
+import json
+import re
+
+
+fc_valid = True
+fg_valid = True
+ccc_valid = True
+nn_valid = True
+cspv_valid = True
+fc_dict = {}
+fg_dict = {}
+ccc_dict = {}
+beam_sources = {'K': '', 'L': ''}
+beam_numbers = {'0': '', '1': '', '2': '', '3': '', '4': '', '5': ''}
```

Comment:
If these keys don't have an useful data to map them to, these could instead be tuples or sets.

Reply:
I'm marking this as unresolved because it was neither adjusted nor justified

Reply:
Oops, I'm going to try to do git updates in either Moba or vs code instead of both. Lost this fix at some point.

Reply:
Changed to tuples and updated the try-except.